### PR TITLE
Migrate from gometalinter.v2 to golangci-lint

### PR DESCRIPTION
**Pull request by thrasher-**
# Description

Due to gometalinter now being EOL, this PR migrates us over to golangci-linter. The benefits are that it's quicker, supports more linters and is easily customisable by a config file. To start, 19 linters have been enabled but over time we will aim to add more.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Travis and make check

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally and on Travis with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/ticker/ticker.go [hunk 1]
```diff
@@ -93,7 +93,8 @@ func GetTickerByExchange(exchange string) (*Ticker, error) {
 	defer m.Unlock()
 	for _, y := range Tickers {
 		if y.ExchangeName == exchange {
-			return &y, nil
+			t := y
```
Comment (shazbert) [inline exchanges/ticker/ticker.go]: Why is this done?

File: config/config.go [hunk 2]
```diff
@@ -257,20 +252,20 @@ func (c *Config) GetCurrencyConfig() CurrencyConfig {
...
 // for depositing funds
-func (c *Config) GetExchangeBankAccounts(exchangeName string, depositingCurrency string) (BankAccount, error) {
+func (c *Config) GetExchangeBankAccounts(exchangeName, depositingCurrency string) (BankAccount, error) {
 	m.Lock()
 	defer m.Unlock()
 
-	for _, exch := range c.Exchanges {
-		if exch.Name == exchangeName {
-			for _, account := range exch.BankAccounts {
+	for x := range c.Exchanges {
```
Comment (gloriousCode) [inline config/config.go]: Is iterating on the index vs putting it in a variable here better practice, more efficient or both?

Comment (thrasher-) [reply config/config.go]: An index or pointer is the most efficient, doing it the original way allocates an exchange config object per iteration.

**Comment by shazbert** [comment]
tACK

**Pull request merged**